code/mathematical_algorithms/src/freivalds_algorithm .cpp

```diff
@@ -0,0 +1,104 @@
+/*
+  Implementation of Freivalds’ algorithm in cpp
+  Part of Cosmos by OpenGenus Foundation
+ */
+
+#include <iostream>
+#include <stdio.h>
+#include <stdlib.h>
+using namespace std;
+int main(int argc, char **argv)
+{
+    cout << "Enter the dimension of the matrices: " << endl;
+    int n;
+    cin >> n;
+    cout << "Enter the 1st matrix: " << endl;
+    double a[n][n];
```

Comment:
Use `std::vector`, as variable length arrays (VLAs), are not part of standard C++.